pkg/gui/commits_panel.go

```diff
@@ -462,21 +462,30 @@ func (gui *Gui) handleCommitPick() error {
 }
 
 func (gui *Gui) handleCommitRevert() error {
-	if ok, err := gui.validateNotInFilterMode(); err != nil || !ok {
-		return err
-	}
-
 	commit := gui.getSelectedLocalCommit()
+	return gui.ask(askOpts{
+		title: gui.Tr.Actions.RevertCommit,
+		prompt: utils.ResolvePlaceholderString(
+			gui.Tr.ConfirmRevertCommit,
+			map[string]string{
+				"selectedCommit": commit.ShortSha(),
+			}),
+		handleConfirm: func() error {
+			if ok, err := gui.validateNotInFilterMode(); err != nil || !ok {
+				return err
+			}
 
-	if commit.IsMerge() {
-		return gui.createRevertMergeCommitMenu(commit)
-	} else {
-		gui.logAction(gui.Tr.Actions.RevertCommit)
-		if err := gui.Git.Commit.Revert(commit.Sha); err != nil {
-			return gui.surfaceError(err)
-		}
-		return gui.afterRevertCommit()
-	}
+			if commit.IsMerge() {
+				return gui.createRevertMergeCommitMenu(commit)
```

Comment:
When menus are involved I prefer letting the user go through the menu process first, and then having a confirmation at the end. Can we do that here too?

Reply:
I have made changes to ask for confirmation before calling `Git.Commit.RevertMerge` in `createRevertMergeCommitMenu`

For now I have hardcoded the confirmation message if this change looks good I can make changes to translate the message

Reply:
Sorry for doing this to you but after discussion in https://github.com/jesseduffield/lazygit/issues/1712 I'm thinking the following policy should apply: if pressing a key brings up a menu, no confirmation is required. If pressing a key immediately does something, a confirmation is required. So in this case, we would check if `IsMerge()` is true and if so we would open a menu but would not confirm afterwards. And if it's false then we'll create the confirmation panel.

Lemme know your thoughts

Reply:
It sounds consistent and I feel it will give users some time to think about their actions rather than simply clicking enter out of muscle memory

I have made the changes according to this new design, please let me know your thoughts